draft-oid4vc-haip-sd-jwt-vc.md

```diff
@@ -139,7 +139,42 @@ Note: Issuers should be mindful of how long the usage of the refresh token is al
 
 ### Wallet Attestation Schema {#wallet-attestation-schema}
 
-[Section 3.1 of wallet attestation draft would define the basics, and this profile will define the details.]
+Wallets MUST use attestations following the definition given in [!I-D.ietf-looker-key-attestation-client-authentication].
+
+In addition to the claims, the Wallet Attestation MUST contain the following claims:
+
+* `key_type`: this claim asserts the security mechanism the wallet can use to manage private keys. This capability is based on the capabilities of the execution environent of the wallet, this might be a secure element (in case of a wallet residing on a smartphone) or a Cloud-HSM (in case of a cloud wallet). This specification defines the following values for `key_type`: `Software`, `TEE`, `Strongbox`, `Secure Enclave`, and `Secure Element`.
+* `user_authentication`: this claim asserts the security mechanism the wallet can use to authenticate access to private keys. This specification defines the following values for `user_authentication`: `System-Biometry`, `System-PIN`, `Internal-Biometry`, `Internal-PIN`, and `SecureElement-PIN`.
+
+These additional claims inform the issuer about the security capabilities of the wallet and allows the issuer to refuse credential issuance if the achievble security level of a certain wallet does not fulfil the issuer's requirements.
+
+Wallet attestions MUST support web-based key resolution as defined in Section 5 of [@!I-D.terbu-sd-jwt-vc]. The JOSE header `kid` MUST be used to identify the respective key.
+
+This is an example of a wallet attestation:
+
+```json
+{
+  "typ": "wallet-attestation+jwt",
+  "alg": "ES256",
+  "kid": "1"
+}
+.
+{
+  "iss": "https://wallet.example.com",
+  "sub": "https://wallet.example.com",
```

Comment:
```suggestion
  "sub": "https://wallet.example.com/version/1.0.0",
```

Reply:
Can you please explain the rationale for the path extension?

Reply:
With regard to eIDAS terms: iss describes the Wallet Solution Provider and sub descibes the specific Wallet Solution of a Wallet Instance

Reply:
Hmm, the string `https://wallet.example.com/version/1.0.0` gives me the impression that there are THREE concepts. Namely,

1. [**per-application identifier**] The traditional client ID assigned to the wallet application. All versions and all per-device instances of the same wallet application use the same client ID.
2. [**per-version identifier**] The version of the wallet application. The string `https://wallet.example.com/version/1.0.0` implies this. The same version of the wallet application may be installed onto multiple devices.
3. [**per-device identifier**] Per-device instance of the wallet application. A key pair generated on the device is only for the per-device instance.

If these three concepts are competing for a single claim, `sub`, then reaching an agreement that satisfies everyone would be impossible. It would be necessary to create new claims as needed.

Reply:
the version is not of interest for an issuer (nor a RP ...) since it should not collect all the changelog/release notes about all the wallet solutions, so that information is useless

Reply:
please reject the version suggestion, see comment above